litestar/response/template.py

```diff
@@ -62,6 +62,9 @@ def __init__(
                 the media type based on the template name. If this fails, fall back to ``text/plain``.
             status_code: A value for the response HTTP status code.
         """
+        if not (template_name or template_str):
+            raise ValueError("Either template_name or template_str must be provided.")
+
```

Comment:
moved this from the create asgi response method to here as better to raise it asap, and consistent with having the "not both" check in here as well.

I started to experiment with using `overload` to type this method so that type checkers will pick up where both are not `None`, but prob getting carried away.